test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -146,6 +146,25 @@ call_log_manager::process_call_log_entry(json call_log_entry)
             }
             break;
         }
+        case api_method::begin_transaction: {
+            const std::string session_id = call_log_entry["session_id"].get<std::string>();
+
+            /*
+             * We should not begin transactions on sessions with an ID that does not exist in the
+             * session map.
+             */
+            if (_session_map.find(session_id) == _session_map.end()) {
+                std::cerr << "Could not begin transaction, session does not exist (session ID: "
+                          << session_id << ")" << std::endl;
+                break;
+            }
+
+            /* Get the session from the session map. */
+            session_simulator *session = _session_map.at(session_id);
+
+            if (!session->begin_transaction())
+                break;
```

Comment:
We should break outside the if statement too?

Reply:
Ah true, I guess we can just leave this if then and handle it when we figure out error handling?

Reply:
yup makes sense.